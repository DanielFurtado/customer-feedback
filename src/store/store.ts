import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
