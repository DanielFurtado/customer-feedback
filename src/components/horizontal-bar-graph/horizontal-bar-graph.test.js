import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import HorizontalBarGraph from './horizontal-bar-graph';

describe(('Feedback Results'), () => {
  const rating = {
    id: 123,
    barColour: '#cccccc',
    label: '5 star',
    startTime: '0.1s',
    percentage: '100'
  };

  it('renders', () => {
    let component = renderer.create(
      <Provider store={store}>
        <Router>
          <HorizontalBarGraph rating={rating} />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct label and percentage', () => {
    render(<HorizontalBarGraph rating={rating} />);
    expect(screen.getByText('5 star')).toBeInTheDocument();
    expect(screen.getByText('100%')).toBeInTheDocument();
  });
});