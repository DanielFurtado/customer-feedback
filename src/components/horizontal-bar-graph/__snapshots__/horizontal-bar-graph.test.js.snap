// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Results renders 1`] = `
<div
  className="sc-eDvSVe jXVCZZ"
>
  <div
    className="sc-dkrFOg fOeNmb"
  >
    5 star
  </div>
  <div
    className="sc-gswNZR kCONEr"
  >
    <div
      className="sc-jSUZER ZGKXo"
    />
  </div>
  <div
    className="sc-hLBbgP iTDVEa"
  >
    100
    %
  </div>
</div>
`;
